caproto/sync/repeater.py

```diff
@@ -127,20 +136,19 @@ def _run_repeater(server_sock, bind_addr):
                 remove_clients = list(check_clients(clients, skip=port))
                 _update_all(clients, servers, remove_clients=remove_clients)
                 # Do not broadcast registration requests to other clients
-                break
-        else:
-            bytes_to_broadcast = b''.join(bytes(cmd) for cmd in commands)
-            to_remove = []
-            for other_port, other_host in clients.items():
-                if other_port != port:
-                    try:
-                        server_sock.sendto(bytes_to_broadcast, (other_host,
-                                                                other_port))
-                    except Exception as ex:
-                        to_remove.append((other_host, other_port))
-
-            if to_remove:
-                _update_all(clients, servers, remove_clients=to_remove)
+                commands.remove(command)
```

Comment:
I have a slight preference for building a new list/deque of commands that should be set out rather than mutating the input.  I am not 100% sure on the equality semantics of commands and worry this might introduce weird corner cases.  Also this is a linear search + a bucket load of shifts,  suspect build a new dequeue would be faster.

Could also pre-allocate a list of the maximum length full on `None`, fill it in and then slice to only what you need below.

*end premature optomization*